Fix build failure with GCC 15+ C++ compiler due to incorrect assumptions about _Bool in C++ mode.

The original fallback logic in syscfg.h attempts to define _Bool and map bool to it when stdbool.h is not available.
 However, _Bool is a C-only type introduced in C99 and does not exist in C++, where bool is a built-in type.

When compiling C++ sources with GCC 15+, this results in a fatal error:

error: '_Bool' does not name a type

--- gnuplot-6.0.3/src/syscfg.h
+++ gnuplot-6.0.3/src/syscfg.h
@@ -378,17 +378,15 @@ #endif
 #if HAVE_STDBOOL_H
 # include <stdbool.h>
 #else
-# if ! HAVE__BOOL
-#  ifdef __cplusplus
-typedef bool _Bool;
-#  else
+# ifndef __cplusplus
+#  if ! HAVE__BOOL
 typedef unsigned char _Bool;
 #  endif
+#  define bool _Bool
+#  define false 0
+#  define true 1
+#  define __bool_true_false_are_defined 1
 # endif
-# define bool _Bool
-# define false 0
-# define true 1
-# define __bool_true_false_are_defined 1
 #endif
 
 /* May or may not fix a problem reported for Sun Studio compilers */
